### PR TITLE
feat(ui): ansi and html in logs is now displayed (#3166)

### PR DESCRIPTION
**Pull request by bnjjj**
Signed-off-by: Benjamin Coenen <benjamin.coenen@corp.ovh.com>

1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: ui/src/app/shared/pipes/safeHtml.pipe.ts [hunk 1]
```diff
@@ -1,10 +1,23 @@
 import {Pipe, PipeTransform} from '@angular/core';
 import {DomSanitizer} from '@angular/platform-browser';
+declare var sanitizeHtml: any;
 
 @Pipe({ name: 'safeHtml'})
 export class SafeHtmlPipe implements PipeTransform  {
   constructor(private sanitized: DomSanitizer) {}
-  transform(value) {
-    return this.sanitized.bypassSecurityTrustHtml(value);
+  transform(value: string, sanitize: boolean) {
```
Comment (richardlt) [inline ui/src/app/shared/pipes/safeHtml.pipe.ts]: Create unit test for `<script>alert("test")</script>` and `<div onclick="alert('test')"></div>`

File: ui/src/app/shared/pipes/safeHtml.pipe.ts [hunk 2]
```diff
@@ -1,10 +1,23 @@
...
+    if (!sanitize) {
```
Comment (richardlt) [inline ui/src/app/shared/pipes/safeHtml.pipe.ts]: Maybe sanitize should not be optional for safeHtml

Comment (bnjjj) [reply ui/src/app/shared/pipes/safeHtml.pipe.ts]: The main goal here is to not do the process twice. In the ansi_up lib when we active the escape html it already do that. It's just for perf. But I'll change the default behavior in order to sanitize by default

File: ui/src/assets/i18n/fr.json [hunk 3]
```diff
@@ -1015,7 +1015,8 @@
   "workflow_notification_form": "Ajouter une notification",
   "workflow_notification_copy": "Copié",
   "workflow_logs_all": "Afficher tous les logs",
-  "workflow_logs_pretty": "Afficher les logs avec interpreteur ANSI et HTML",
+  "workflow_logs_pretty_ansi": "Afficher les logs avec interpreteur ANSI",
```
Comment (sguiheux) [inline ui/src/assets/i18n/fr.json]:
```suggestion
  "workflow_logs_pretty_ansi": "Afficher les logs avec l'interpréteur ANSI",
```

File: ui/src/assets/i18n/fr.json [hunk 4]
```diff
@@ -1015,7 +1015,8 @@
...
+  "workflow_logs_pretty_html": "Afficher les logs avec interpreteur HTML",
```
Comment (sguiheux) [inline ui/src/assets/i18n/fr.json]:
```suggestion
  "workflow_logs_pretty_html": "Afficher les logs avec l'interpréteur HTML",
```

**Pull request merged**